app/Http/Controllers/AdminDash.php

```diff
@@ -2077,4 +2073,38 @@ public function saveLiveEventInfo(Request $request) {
 
         return redirect('/dashboard/admin/live')->with('success', 'The live event info has been updated successfully.');
     }
+
+    public function removeSoloCertifications(Request $request) {
+        if (!Auth::user()->isAbleTo('roster')) {
+            return redirect('/dashboard/controllers/roster')->with('error', 'Insufficient permissions.');
+        }
+        $user = User::find($request->id);
+        $user->twr_solo_fields = '';
+        $user->twr_solo_expires = '';
+        $positions = ['twr','app','ctr'];
+        foreach ($positions as $position) {
+            if ($user[$position] == $user->getMagicNumber('SOLO_CERTIFICATION')) {
+                $user[$position] = $user->getMagicNumber('UNCERTIFIED');
+            }
+        }
+        $user->save();
+        $client = new Client(['exceptions' => false]);
+        $response = $client->request('GET', Config::get('vatusa.base').'/v2/solo'.'?apikey='.Config::get('vatusa.api_key'));
+        $result = $response->getStatusCode();
+        if ($result == '200') {
+            $all_vatusa_solo_certs = json_decode($response->getBody());
+            foreach ($all_vatusa_solo_certs->data as $solo_cert) {
+                if ($solo_cert->cid == $user->id && in_array($solo_cert->position, $user->soloPositions())) {
+                    (new Client())->request('DELETE', Config::get('vatusa.base').'/v2/solo'.'?apikey='.Config::get('vatusa.api_key').'&id='.$solo_cert->id, ['http_errors' => false]);
+                }
+            }
```

Comment:
This is fine for what we have now, but in the future we may want to consider storing this solo cert ID for this removal purpose. Because looping through every issued solo cert seems very inefficient. It also adds an extra API call which will slow things down.